src/Api/Persistent.hs

```diff
@@ -159,6 +160,14 @@ getSpaces = getDb dbSpaces
 getIdeas :: Persist [Idea]
 getIdeas = getDb dbIdeas
 
+-- | If idea space already exists, return 'False'.  Otherwise, create it and return 'True'.
+addIdeaSpace :: IdeaSpace -> Persist Bool
```

Comment:
The runPersist should have `Persist :~> ErrorT PersistError IO` instead of `Persist :~> IO` . And with that you can throw an error if the entity exist already. Or connection error happened, etc.